src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -446,6 +484,7 @@ private void completeJob() {
         this.numsOfOutput = 0;
         this.isComplete = true;
         this.playersFollowingCurrentCraft.clear();
+        this.craftCompleteListeners = initializeDefaultOnCompleteListener();
```

Comment:
To continue logic of per-craft subscription other types of listeners (status update/cancel) also should be reset here. Since craft is finished successfully.

Reply:
They are added.